### ISSUE TITLE
`@overload` Tags do not receive noImplicitAny errors on missing return types

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#52514]
[Playground Link](https://www.typescriptlang.org/play?strictNullChecks=true&ts=5.0.0-dev.20230130&filetype=js#code/PTAEAEBcGcFoGMAWBTeBrAUB4AqHHQcIB7AN2QCcAbYgQwBMCjwAHWi2gW1AG8A7AK6cARpQC+oAB5NgWXPkIly1Oo0Wt2XXtEgUAlnwDmE6YVnY8TCGw7ceO-UdAAfUIJHipV8BWSQBFHzQ2roGhi5uQqIUYjIYAGYCfPCQesR8oHr0ABSSAJS8BKDFvv6BUgDcGLEYWdkAjABMAMx5QA)

```ts
// @ts-check

/**
 * @overload
 * @param {number} x
 */

/**
 * @overload
 * @param {string} x
 */

/**
 * @param {string | number} x
 * @returns {string | number}
 */
function id(x) {
    return x;
}

export let a = id(123);
export let b = id("hello");
```

Here, the first two signatures of `id` are missing an `@return`/`@returns` tag. That means that they're implicitly `any`.

Under `noImplicitAny`, this example should issue an error on both `@overload` tags saying that they implicitly return the type `any`.
